### ISSUE TITLE
installExternalPackages fails if package does not have dependencies or devDependencies

### ISSUE THREAD

**Issue by ianwremmel** [kittens/lerna#103]
These lines assume `dependencies` and `devDependencies` are both defined objects.

```javascript
  var deps = Object.keys(pkg.pkg.dependencies);
  var devDeps = Object.keys(pkg.pkg.devDependencies);
```

**Comment by hzoo** [comment]
Good point - what version is this for?

**Comment by ianwremmel** [comment]
I'm trying out 2.0.0-beta2, but it seems to have some perf issues so I don't think I'm ready make the switch yet.

**Comment by thejameskyle** [comment]
What perf issues? It's significantly faster for me

**Comment by ianwremmel** [comment]
I've been using [my fork](https://github.com/ianwremmel/lerna/tree/link-order) for about a month and lerna bootstrap takes about 40 seconds when all modules are installed. With 2.0.0-beta2, I aborted lerna bootstrap after 4 minutes and the progress indicator looked like about a third of my packages had been visited.

**Comment by thejameskyle** [comment]
Do you have a repo up somewhere that shows this?

**Comment by ianwremmel** [comment]
unfortunately, no. it's an internal project that won't be public for another few months.

**Comment by ianwremmel** [comment]
I'll see what I can do about getting more info for the perf issue, but in the mean time (1) should I open a new issue and (2) is it possible that it's an npm 2 vs npm 3? I'm using npm 2.

**Comment by thejameskyle** [comment]
I might be related to your version of npm, I'll do more testing with npm 2 since I use npm 3 for the most part

**Comment by thejameskyle** [comment]
The original issue has been fixed in #105 

@ianwremmel if you can demonstrate a perf issue please open a new issue

**Comment by ianwremmel** [comment]
Will do, thanks for the quick responses!
